### PR TITLE
Adding 'mirai' package

### PR DESCRIPTION
**Pull request by shikokuchuo**
As suggested by @HenrikBengtsson at shikokuchuo/mirai#17. Thanks.

### CODE REVIEW

File: HighPerformanceComputing.md [hunk 1]
```diff
@@ -79,6 +79,10 @@ for later R releases.
     supported. Parallel map-reduce calls via the future framework are provided by packages
     `r pkg("future.apply")` for parallel versions of base-R apply functions, and
     `r pkg("furrr")` for parallel versions of purrr fuctions.
+-   The `r pkg("mirai")` package is a minimalist, highly-scalable framework for asynchronous
+    evaluation, allowing concurrent code execution locally or across the network, implementing
+    futures which automatically resolve upon completion. Uses the lightweight NNG C library as
+    backend through the high-performance `r pkg("nanonext")` binding.
```
Comment (eddelbuettel) [inline HighPerformanceComputing.md]: Please use full sentence.

Please try to shorten it as well.  Most entries try to get by with two lines.  

Please move it to the end of the section as section grow chronologically.

**Review by eddelbuettel — CHANGES_REQUESTED** [review]

File: HighPerformanceComputing.md (hunk 1 above)
Comment (shikokuchuo) [reply HighPerformanceComputing.md]: Have amended the PR accordingly. Thanks.

File: HighPerformanceComputing.md [hunk 2]
```diff
@@ -96,7 +92,9 @@ for later R releases.
 -   The `r pkg("parSim")` package can perform simulation studies using one or multiple cores, both
     locally and on HPC clusters.
 -   The `r pkg("qsub")` package can submit commands to run on gridengine clusters.
-
+-   The `r pkg("mirai")` package is a minimalist framework for local or distributed asynchronous
+    code evaluation, implementing futures which automatically resolve upon completion, built on the
+    high-performance `r pkg("nanonext")` NNG C library binding.
 ### Parallel computing: Implicit parallelism
```
Comment (eddelbuettel) [inline HighPerformanceComputing.md:99]: Very nice, thanks,  Do we need to define / expand what NNG stands for? 

And if I may, a linebreak please between what is shown here as lines 97 and 98.   

(I leave this one up to you as my mother tongue was also one in which sentence tend to run too long but I think we should have a full stop in the middle after 'code evaluation.' to continue with 'It implements ...'   But that is not mandatory.  )

**Review by eddelbuettel — CHANGES_REQUESTED** [review]

File: HighPerformanceComputing.md (hunk 2 above)
Comment (shikokuchuo) [reply HighPerformanceComputing.md:99]: I spotted the missing line break and added that back in the following commit.

I have added that NNG is a messaging library, which I hope makes its use self-evident. It is simply known as 'libnng' and I don't think expanding it to 'Nanomsg Next Gen' helps much, you'd probably have to go all the way to say that it is a successor to ZeroMQ!

As for style, I prefer to leave it as a long sentence - I think it reads fine.

Comment (eddelbuettel) [reply HighPerformanceComputing.md:99]: Yes _I_ know what NNG as I follow your packages from a distance via the CRANBerries update.  Readers may or may not know, but they do have the package link.

GitHub may have shown me the diff one commit behind your flow,

**Review by eddelbuettel — APPROVED** [review]
Looks good, thanks.

**Pull request merged**